ozaria/site/components/common/LayoutAspectRatioContainer.vue

```diff
@@ -0,0 +1,66 @@
+<script>
+  export default {
+    props: {
+      aspectRatio: {
+        required: true,
+        type: Number
+      }
+    },
+
+    data () {
+      return {
+        parentWidth: 0,
+        parentHeight: 0
+      }
+    },
+
+    computed: {
+      myHeight () {
+        return Math.min(this.parentWidth / this.aspectRatio, this.parentHeight)
+      },
+
+      myWidth () {
+        return this.myHeight * this.aspectRatio
+      }
+    },
+
+    mounted () {
+      window.addEventListener('resize', this.onResize)
+      this.onResize()
+    },
+
+    beforeDestroy () {
+      window.removeEventListener('resize', this.onResize)
+    },
+
+    methods: {
+      onResize (e) {
+        const parent = this.$refs.el.parentElement
+        if (!parent) {
+          throw new Error('Element does not have parent')
+        }
+
+        const boundingRect = parent.getBoundingClientRect()
+
+        if (boundingRect) {
+          this.parentWidth = boundingRect.width
+          this.parentHeight = boundingRect.height
+        } else {
+          this.parentWidth = parent.clientWidth
+          this.parentHeight = parent.clientHeight
+        }
+
+        this.$emit('resize')
```

Comment:
Is it useful for this to emit the width and height?
So that a system like cinematics can resize the canvas manually.

Reply:
I went back and forth on this.  I opted to not send in the event because the width and height available here are not actually useful to children containers.  I also considered emitting an event with the calculated width and height however I am not sure exactly how events work in relation to actual resizing / page draws.

To keep things consistent and clear I opted to simply send the resize and put the onus of measuring on the even consumer.  Alternatively the consumer could read myWidth / myHeight via.  We could give those better names as well.  I think either of those scenarios make it easier to reason about what the width / height actually is.

Reply:
Sounds good. Once this is merged I'll have a play with cinematics to see if I can get the resize working with the padding. (I was having some issue locally)